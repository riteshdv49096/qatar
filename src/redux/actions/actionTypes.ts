export const FETCH_BLOG_POSTS = 'FETCH_BLOG_POSTS';
export const FETCH_BLOG_POSTS_SUCCESS = 'FETCH_BLOG_POSTS_SUCCESS';
export const FETCH_BLOG_POSTS_FAILURE = 'FETCH_BLOG_POSTS_FAILURE';

export const EDIT_BLOG_POST = 'EDIT_BLOG_POST';
export const EDIT_BLOG_POST_SUCCESS = 'EDIT_BLOG_POST_SUCCESS';
export const EDIT_BLOG_POST_FAILURE = 'EDIT_BLOG_POST_FAILURE';

export const FETCH_USER = 'FETCH_USER';
export const FETCH_USER_SUCCESS = 'FETCH_USER_SUCCESS';
export const FETCH_USER_FAILURE = 'FETCH_USER_FAILURE';
