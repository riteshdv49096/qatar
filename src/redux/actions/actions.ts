import axios from 'axios';
import {generateRandomNumber} from '../../utils/Utils';
import {
  FETCH_BLOG_POSTS,
  FETCH_BLOG_POSTS_SUCCESS,
  FETCH_BLOG_POSTS_FAILURE,
  EDIT_BLOG_POST,
  EDIT_BLOG_POST_SUCCESS,
  EDIT_BLOG_POST_FAILURE,
  FETCH_USER,
  FETCH_USER_SUCCESS,
  FETCH_USER_FAILURE,
} from './actionTypes';
import {BASE_URL} from '../../utils/Constants';

const fetchUser = () => {
  return {
    type: FETCH_USER,
  };
};

const fetchUserSuccess = (data: any) => {
  return {
    type: FETCH_USER_SUCCESS,
    payload: data,
  };
};

const fetchUserFailure = (error: any) => {
  return {
    type: FETCH_USER_FAILURE,
    error: error.message,
  };
};

/**
 * This function get the user data
 * @returns user data
 */
export const fetchUserData = () => {
  return (dispatch: any) => {
    dispatch(fetchUser());
    axios
      .get(`${BASE_URL}/users/${generateRandomNumber()}`)
      .then(res => {
        dispatch(fetchUserSuccess(res.data));
        dispatch(fetchBlogs(res.data.id));
      })
      .catch(error => {
        fetchUserFailure(error.message);
      });
  };
};

const fetchBlogPosts = () => {
  return {
    type: FETCH_BLOG_POSTS,
  };
};

const fetchBlogPostsSuccess = (data: any) => {
  return {
    type: FETCH_BLOG_POSTS_SUCCESS,
    payload: data,
  };
};

const fetchBlogPostsFailure = (error: any) => {
  return {
    type: FETCH_BLOG_POSTS_FAILURE,
    payload: error,
  };
};

/**
 * This function return the blogs from api
 * @param userId number
 * @returns 
 */
export const fetchBlogs = (userId: any) => {
  return (dispatch: any) => {
    dispatch(fetchBlogPosts());
    axios
      .get(`${BASE_URL}/users/${userId}/posts`)
      .then(res => {
        dispatch(fetchBlogPostsSuccess(res.data));
      })
      .catch(error => {
        dispatch(fetchBlogPostsFailure(error.message));
      });
  };
};

const editBlogs = () => {
  return {
    type: EDIT_BLOG_POST,
  };
};

const editBlogPostSuccess = (data: any) => {
  return {
    type: EDIT_BLOG_POST_SUCCESS,
    payload: data,
  };
};

const editBlogPostFailure = (error: any) => {
  return {
    type: EDIT_BLOG_POST_FAILURE,
    payload: error,
  };
};

/**
 * This function update the post using put method
 * 
 * @param post object
 * @returns 
 */
export const editBlogPost = (post: any) => {
  return (dispatch: any) => {
    dispatch(editBlogs());
    axios
      .put(`${BASE_URL}/posts/${post.id}`, post)
      .then(res => {
        dispatch(editBlogPostSuccess(res.data));
      })
      .catch(error => {
        dispatch(editBlogPostFailure(error.message));
      });
  };
};
