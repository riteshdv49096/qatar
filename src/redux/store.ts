import { applyMiddleware, legacy_createStore } from 'redux';
import rootReducer from './reducers/reducer';
import thunk from 'redux-thunk';

const configureStore = legacy_createStore(rootReducer, applyMiddleware(thunk));

export default configureStore;
