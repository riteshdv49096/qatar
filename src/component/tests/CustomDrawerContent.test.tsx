import { SectionList, Text } from 'react-native';
import { create } from 'react-test-renderer';
import { fetchUserData } from '../../redux/actions/actions';
import { userAPI } from '../../redux/actions/actions';
import { FETCH_USER, FETCH_USER_SUCCESS } from '../../redux/actions/actionTypes';
import store from '../../redux/store';

describe('validate the navigation drawer', () => {
    it('validate drawer menus', () => {
        const tree = create(
            <SectionList
                sections={[
                    { title: 'Dashboard', data: ['Overview', 'Calender', 'Schedule Action', 'Live Alert'] },
                    { title: 'Blogs', data: ['Blogs', 'AllBlogs', 'Archive'] },
                ]}
                keyExtractor={(item, index) => item + index}
                renderItem={({ item }) => <Text>{item}</Text>}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
