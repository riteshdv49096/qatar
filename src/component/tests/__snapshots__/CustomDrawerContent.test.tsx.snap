// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate the navigation drawer validate drawer menus 1`] = `
<RCTScrollView
  data={
    [
      {
        "data": [
          "Overview",
          "Calender",
          "Schedule Action",
          "Live Alert",
        ],
        "title": "Dashboard",
      },
      {
        "data": [
          "Blogs",
          "AllBlogs",
          "Archive",
        ],
        "title": "Blogs",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    [
      0,
      6,
    ]
  }
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Overview
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Calender
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Schedule Action
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Live Alert
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Blogs
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        AllBlogs
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Archive
      </Text>
    </View>
    <View
      style={
        {
          "height": 0,
        }
      }
    />
  </View>
</RCTScrollView>
`;
